hardware/arduino/avr/cores/arduino/CDC.cpp

```diff
@@ -196,13 +197,19 @@ size_t Serial_::write(const uint8_t *buffer, size_t size)
 // setup() before printing to ensure that an application on the host is
 // actually ready to receive and display the data.
 // We add a short delay before returning to fix a bug observed by Federico
-// where the port is configured (lineState != 0) but not quite opened.
+// where the port is configured (lineState != 0) but not quite opened.  This
+// only happens when the line state transitions from ==0 to >0 otherwise you
+// get a nasty delay when one isn't needed.
 Serial_::operator bool() {
-	bool result = false;
-	if (_usbLineInfo.lineState > 0) 
-		result = true;
-	delay(10);
-	return result;
+    bool res = false;
```

Comment:
I'd not rename it to res. Res sounds weird to me. Please keep result. The rest seems like a good idea to me.